### ISSUE TITLE
NumericInput button icons should have <title> for a11y

### ISSUE THREAD

**Issue by adidahiya** [palantir/blueprint#4527]
#### Feature request

There are some cases where BP's built-in components could use better accessibility for icons. We decided a while ago in https://github.com/palantir/blueprint/issues/3214 that not _all_ icons should get tooltips, but in cases like this where there is no label for the icon, I think it makes sense to add one by default.

#### Expected behavior

![image](https://user-images.githubusercontent.com/723999/108098529-7dca3d80-7051-11eb-938d-0003579a24fb.png)
